# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/ps_text_run.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/ps_text_run.glsl
--- a/gfx/wr/webrender/res/ps_text_run.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/ps_text_run.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -4,17 +4,17 @@
 
 #include shared,prim_shared
 
-flat varying vec4 v_color;
-flat varying vec3 v_mask_swizzle;
+flat varying mediump vec4 v_color;
+flat varying mediump vec3 v_mask_swizzle;
 // Normalized bounds of the source image in the texture.
-flat varying vec4 v_uv_bounds;
+flat varying highp vec4 v_uv_bounds;
 
 // Interpolated UV coordinates to sample.
-varying vec2 v_uv;
+varying highp vec2 v_uv;
 
 
 #if defined(WR_FEATURE_GLYPH_TRANSFORM) && !defined(SWGL_CLIP_DIST)
-varying vec4 v_uv_clip;
+varying highp vec4 v_uv_clip;
 #endif
 
 #ifdef WR_VERTEX_SHADER